newton/_src/utils/benchmark.py

```
@@ -13,9 +13,148 @@
+
+      with EventTracer() as tracer:
+        my_warp_function(...)
+        print(tracer.trace())
+    """
+
+    _STACK = None
+    _active_instance = None
+
+    def __new__(cls, enabled):
+        if EventTracer._active_instance is not None and enabled:
+            raise ValueError("only one EventTracer can run at a time")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)